channel.go

```diff
@@ -0,0 +1,340 @@
+package cli
+
+import (
+	"encoding/json"
+	"time"
+
+	stream "github.com/GetStream/stream-chat-go/v5"
+	"github.com/cheynewallace/tabby"
+	"github.com/urfave/cli/v2"
+)
+
+func NewChannelCmd(config *Config) *cli.Command {
+	return &cli.Command{
+		Name:        "channel",
+		Usage:       "Manage channels.",
+		Description: "Manage channels.",
+		Subcommands: []*cli.Command{
+			getChannelCmd(config),
+			newListChannelsCmd(config),
+			newCreateChannelCmd(config),
+			newDeleteChannelCmd(config),
+			newUpdateChannelCmd(config),
+		},
+	}
+}
+
+func getChannelCmd(config *Config) *cli.Command {
+	return &cli.Command{
+		Name:        "get",
+		Usage:       "Get a channel by channel type and channel name.",
+		Description: "Get a channel by channel type and channel name.",
+		Flags: []cli.Flag{
+			&cli.StringFlag{
+				Name:     "channel-type",
+				Aliases:  []string{"t"},
+				Usage:    "The type of the channel. Such as 'messaging' or 'livestream'.",
+				Required: true,
+			},
+			&cli.StringFlag{
+				Name:     "channel-name",
+				Aliases:  []string{"n"},
+				Usage:    "The name of the channel.",
+				Required: true,
+			},
+			&cli.BoolFlag{
+				Name:     "json",
+				Usage:    "Print the raw JSON representation of the API response.",
+				Required: false,
+			},
+		},
+		Action: func(ctx *cli.Context) error {
+			r, err := getOrCreateChannel(config, ctx)
+
+			if err != nil {
+				return err
+			}
+
+			t := tabby.New()
+			t.AddHeader("CID", "Member count", "Created By", "Created At", "Updated At", "Extra Data")
+			t.AddLine(r.Channel.CID,
+				r.Channel.MemberCount,
+				r.Channel.CreatedBy.ID,
+				r.Channel.CreatedAt.Format(time.RFC822),
+				r.Channel.UpdatedAt.Format(time.RFC822),
+				r.Channel.ExtraData)
+
+			if ctx.Bool("json") {
+				PrintRawJson(r)
+			} else {
+				t.Print()
+			}
+
+			return nil
+		},
+	}
+}
+
+func newCreateChannelCmd(config *Config) *cli.Command {
+	return &cli.Command{
+		Name:      "create",
+		Usage:     "Create a new channel or return an existing one.",
+		UsageText: "stream-cli channel create --channel-type [channel-type] --channel-name [channel-name]",
+		Description: "Creates a new channel or returns an existing one if it already exists. " +
+			"If channel name is not provided it will be generated automatically.",
+		Flags: []cli.Flag{
+			&cli.StringFlag{
+				Name:     "channel-type",
+				Aliases:  []string{"t"},
+				Usage:    "The type of the channel. Such as 'messaging' or 'livestream'.",
+				Required: true,
+			},
+			&cli.StringFlag{
+				Name:     "channel-name",
+				Aliases:  []string{"n"},
+				Usage:    "The name of the channel.",
+				Required: true,
+			},
```

Comment:
Let's define these two flags as global flag for all channel's tree commands, so that we don't have to define them for each subcommand

Reply:
good idea

Reply:
won't work, because the List command doesn't need channel id

Reply:
we can either:
- put a default value
- don't make it required and check in each command when it's needed with a Before hook for example (i prefer this one)

Reply:
Additionally that also messes up the mandatory/non-mandatory parameters lists. For example:

<img width="720" alt="image" src="https://user-images.githubusercontent.com/19969687/156172686-2b85868f-3f0f-4dba-9c12-5440653b9218.png">

# versus


<img width="764" alt="image" src="https://user-images.githubusercontent.com/19969687/156173032-1b5c1c9d-3ba5-44d8-84f7-1cecae5f9144.png">

Reply:
btw, is it a conscious decision to call it channel-name instead of channel-id?

Reply:
nope, will fix it

Reply:
I meant "intentional" instead of "conscious", in case it means something else 😆